lib/srv/regular/proxy.go

```diff
@@ -264,8 +277,9 @@ func (t *proxySubsys) Wait() error {
 	return <-t.closeC
 }
 
-// doHandshake allows a proxy server to send additional information (client IP)
-// to an SSH server before establishing a bridge
+// doHandshake allows a proxy server to send additional information (client IP and tracing context)
+// to an SSH server before establishing a bridge.
+// NOTE: Used for compatibility with versions <12.1, before IP propagation through signed PROXY headers was added.
```

Comment:
Do we need to identify and signal deprecation here with a removal version ?

Reply:
Yes we probably should. Considering our support of latest 3 version and promise to work with one version difference, I think it means this should be removed in v16.

Reply:
In that case, I would add a comment stating "DELETE IN 14.0.0" - otherwise there's a good chance this will float around in the codebase for a while :3